src/iotjs_module_buffer.cpp

```diff
@@ -216,7 +216,9 @@ JHANDLER_FUNCTION(Write) {
 
   BufferWrap* buffer_wrap = BufferWrap::FromJBufferBuiltin(*jbuiltin);
 
-  size_t copied = buffer_wrap->Copy(src.data(), 0, length, offset);
+  int length_min = src.size() < length ? src.size() : length;
```

Comment:
What is the behavior of Node.js in such cases? Wouldn't be better to throw an exception if length parameter is bigger than the length of the source?